Purchases/FoundationExtensions/Dictionary+Extensions.swift

```diff
@@ -88,3 +88,23 @@ extension Dictionary {
     }
 
 }
+
+extension Sequence {
+    /// Creates a `Dictionary` with the values in the receiver sequence, and the keys provided by `key`.
+    /// - Precondition: The sequence must not have duplicate keys.
+    func uniqueIndex<Key>(
```

Comment:
I feel like there has to be a better name for this. `uniqueIndex` doesn't seem very discoverable or descriptive enough IMO.

Reply:
I agree. Maybe `dictionary(withKeys:)`? I'm not the best at finding names for these.

Reply:
Changed in #1090.